include/kernel.h

```diff
@@ -4846,61 +4846,6 @@ extern void _arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 }
 #endif
 
-#if defined(CONFIG_CPLUSPLUS) && defined(__cplusplus)
-/*
- * Define new and delete operators.
- * At this moment, the operators do nothing since objects are supposed
- * to be statically allocated.
- */
-inline void operator delete(void *ptr)
```

Comment:
Is `kernel.h` the appropriate place for this? The kernel neither {exports, requires} C++.

Reply:
Before this commit, dummy new/delete was declared in kernel.h. 
Now I moved them into subsys/cpp/new.cpp, which is compiled only when CONFIG_CPLUSPLUS flag is setted.

Reply:
OK, better than before I guess :)
Could I lure you to create another PR (or issue at least) for moving C++ out of `kernel.h`?